### PR TITLE
Consensus refactor

### PR DESCRIPTION
**Pull request by wkoszycki**

### CODE REVIEW

File: src/main/scala/org/constellation/DAO.scala [hunk 1]
```diff
@@ -104,6 +101,16 @@ class DAO() extends NodeData with EdgeDAO with SimpleWalletLike with StrictLoggi
     ipManager = IPManager()
     cluster = Cluster[IO](() => metrics, ipManager, this)
 
+    consensusRemoteSender = new ConsensusRemoteSender[IO]()
+    consensusManager = new ConsensusManager[IO](transactionService,
+                                                concurrentTipService,
```
Comment (marcinwadon) [inline src/main/scala/org/constellation/DAO.scala]: Please format it properly

File: src/main/scala/org/constellation/consensus/ConsensusManager.scala [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+    roundId: RoundId,
+    peers: Set[PeerData],
+    transactionsProposal: LightTransactionsProposal
+  )
+
+  case object NodeNotReady extends ConsensusStartError("Node not ready to start consensus")
+  case object OwnRoundAlreadyInProgress extends ConsensusStartError("Node has already start own consensus")
+
+  class ConsensusStartError(message: String) extends Exception(message)
+  class ConsensusError(val roundId: RoundId, message: String) extends Exception(message)
```
Comment (marcinwadon) [inline src/main/scala/org/constellation/consensus/ConsensusManager.scala]: why not case class?

File: src/main/scala/org/constellation/consensus/Consensus.scala [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+  case class ResolveMajorityCheckpointBlock(roundId: RoundId, stageState: StageState) extends RoundCommand
+
+  case class AcceptMajorityCheckpointBlock(roundId: RoundId) extends RoundCommand
+
+  case class StartTransactionProposal(roundId: RoundId) extends RoundCommand
+
+  case class LightTransactionsProposal(
+    roundId: RoundId,
+    facilitatorId: FacilitatorId,
+    txHashes: Seq[String],
```
[the remark this answers is not in the document]
Comment (wkoszycki) [reply src/main/scala/org/constellation/consensus/Consensus.scala]: Well I agree but its a big change, and we could consider to change it within `CheckpointBlock` . Also Messages could be reconsider.

File: src/main/scala/org/constellation/consensus/ConsensusManager.scala [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+            dao.threadSafeMessageMemPool.activeChannels
+              .get(message.signedMessageData.data.channelId)
+              .foreach(_.release())
+      )
+    )
+
+  def cleanUpLongRunningConsensus: F[Unit] =
+    for {
+      runningRounds <- rounds.getUnsafe
+      currentTime <- Sync[F].delay(System.currentTimeMillis())
```
[the remark this answers is not in the document]
Comment (wkoszycki) [reply src/main/scala/org/constellation/consensus/ConsensusManager.scala]: Is there any difference except syntax sugar ?

File: src/main/scala/org/constellation/consensus/ConsensusManager.scala (hunk 2 above)
Comment (wkoszycki) [reply src/main/scala/org/constellation/consensus/ConsensusManager.scala]: case to case inheritance is prohibited

File: src/main/scala/org/constellation/consensus/ConsensusManager.scala [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+                                         remoteSender: ConsensusRemoteSender[F],
+                                         cluster: Cluster[F],
+                                         dao: DAO,
+                                         config: Config) {
+
+  import ConsensusManager._
+
+  implicit val shadowDAO: DAO = dao
+
+  val logger: SelfAwareStructuredLogger[F] = Slf4jLogger.getLogger[F]
```
[the remark this answers is not in the document]
Comment (wkoszycki) [reply src/main/scala/org/constellation/consensus/ConsensusManager.scala]: When logger is instantiated in different package you are loosing ability to manager log levels per class/package

**Pull request merged**